src/SprykerEco/Zed/Computop/Persistence/ComputopEntityManager.php

```diff
@@ -56,10 +63,144 @@ public function updatePaymentComputopOrderItemPaymentConfirmation(
+    {
+        $computopApiResponseHeaderTransfer = $computopPayuCeeSingleInitResponseTransfer->requireHeader()
+            ->getHeader()
+                ->requireTransId()
+                ->requirePayId()
+                ->requireXid();
+
+        $paymentComputopEntity = $this->getFactory()
+            ->createPaymentComputopQuery()
+            ->filterByTransId($computopApiResponseHeaderTransfer->getTransId())
+            ->findOne();
+
+        if (!$paymentComputopEntity) {
+            return;
+        }
+
+        $paymentComputopEntity
+            ->setPayId($computopApiResponseHeaderTransfer->getPayId())
+            ->setXId($computopApiResponseHeaderTransfer->getXId())
+            ->save();
+
+        $this->savePaymentComputopDetailEntity(
+            $paymentComputopEntity->getSpyPaymentComputopDetail(),
+            $computopPayuCeeSingleInitResponseTransfer
+        );
+
+        $this->savePaymentComputopOrderItems(
+            $paymentComputopEntity->getSpyPaymentComputopOrderItems(),
+            $this->getPaymentStatus($computopPayuCeeSingleInitResponseTransfer)
+        );
+    }
+
+    /**
+     * @param \Orm\Zed\Computop\Persistence\SpyPaymentComputopDetail $paymentComputopDetailEntity
+     * @param \Generated\Shared\Transfer\ComputopPayuCeeSingleInitResponseTransfer $computopPayuCeeSingleInitResponseTransfer
+     *
+     * @return void
+     */
+    protected function savePaymentComputopDetailEntity(
+        SpyPaymentComputopDetail $paymentComputopDetailEntity,
+        ComputopPayuCeeSingleInitResponseTransfer $computopPayuCeeSingleInitResponseTransfer
+    ): void {
+        $paymentComputopDetailEntity->fromArray($computopPayuCeeSingleInitResponseTransfer->toArray());
+        $customerTransactionId = $computopPayuCeeSingleInitResponseTransfer->getCustomerTransactionId();
+        if ($customerTransactionId) {
+            $paymentComputopDetailEntity->setCustomerTransactionId((int)$customerTransactionId);
+        }
+
+        $paymentComputopDetailEntity->save();
+    }
+
+    /**
+     * @param \Orm\Zed\Computop\Persistence\SpyPaymentComputopOrderItem[]|\Propel\Runtime\Collection\ObjectCollection $paymentComputopOrderItemEntities
+     * @param string $paymentStatus
+     *
+     * @return void
+     */
+    protected function savePaymentComputopOrderItems(ObjectCollection $paymentComputopOrderItemEntities, string $paymentStatus): void
+    {
+        foreach ($paymentComputopOrderItemEntities as $paymentComputopOrderItem) {
+            $paymentComputopOrderItem->setStatus($paymentStatus);
+            $paymentComputopOrderItem->save();
+        }
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ComputopNotificationTransfer $computopNotificationTransfer
+     * @param \Orm\Zed\Computop\Persistence\SpyPaymentComputopOrderItem $paymentComputopOrderItemEntity
+     *
+     * @return string
+     */
+    protected function getCurrentOrderItemEntityStatus(
```

Comment:
Not a part of entity manager and it's not a getter but mapper

Reply:
Yes but if I move this method outside to NotificationProcessor it will be major release.